compiler-core/src/language_server/compiler.rs

```diff
@@ -114,6 +114,22 @@ where
             _ = self.sources.insert(module.name.clone(), source);
         }
 
+        // Since cached modules are not recompiled we need to manually add them
+        for (name, module) in self.project_compiler.get_importable_modules() {
```

Comment:
This is loading information about cached modules every time the compiler runs, but they will have never changed. Instead we could do this once when the compiler is created. What do you think?

Reply:
So the issue with that is that importable_modules won't actually have the right list until the first compile when the package loader is first run. For what its worth, https://github.com/gleam-lang/gleam/pull/2861/files/dcabe44c72ffcad5bf528a97dd0dc08adb6952dc#diff-9b6f5e817763c0c820d9ece57897c33972be558d7348018dc916fac80e5d9968R119 this check does prevent the bulk of the work since after the first compile the sources will already be present.

Reply:
I added a trace to the cache loading and confirmed it only goes through the loop on the very first compile which seems reasonable to me? I think the alternative would be to run the package_loader directly which could be doable though might be a bit more of a lift than necessary here?